src/storage/src/hummock/compactor/iterator.rs

```diff
@@ -362,48 +361,22 @@ impl ConcatSstableIterator {
                 .verbose_instrument_await("stream_iter_sstable")
                 .await?;
             let block_metas = &sstable.meta.block_metas;
-            let mut start_index = match seek_key {
-                None => 0,
-                Some(seek_key) => {
-                    // start_index points to the greatest block whose smallest_key <= seek_key.
-                    block_metas
-                        .partition_point(|block| {
-                            seek_key.cmp(&FullKey::decode(&block.smallest_key)) != Ordering::Less
-                        })
-                        .saturating_sub(1)
-                }
-            };
-            let end_index = if self.key_range.right.is_empty() {
-                block_metas.len()
-            } else {
-                block_metas.partition_point(|block| {
-                    KeyComparator::compare_encoded_full_key(
-                        &block.smallest_key,
-                        &self.key_range.right,
-                    ) != Ordering::Greater
-                })
-            };
-            while start_index < end_index {
-                let start_block_table_id = block_metas[start_index].table_id();
-                if self
-                    .existing_table_ids
-                    .contains(&block_metas[start_index].table_id().table_id)
-                {
-                    break;
-                }
-                start_index += &block_metas[(start_index + 1)..]
-                    .partition_point(|block_meta| block_meta.table_id() == start_block_table_id)
-                    + 1;
-            }
-            if start_index >= end_index {
+            let (start_index, end_index) = Self::filter_block_metas(
```

Comment:
How about:
- Passing the ownership of `block_metas` to `filter_block_metas` instead of passing by ref
- Letting `filter_block_metas` returns a `Vec<BlockMeta>` directly

Reply:
1. We can't move the block_meta out of the CacheEntry, cloning the whole block_meta in advance is not necessary (input should be ref)
2. We always need to copy the subset, so it makes sense to return Vec.